src/Interpreters/SyntaxAnalyzer.cpp

```diff
@@ -828,6 +961,9 @@ SyntaxAnalyzerResultPtr SyntaxAnalyzer::analyzeSelect(
         /// GROUP BY injective function elimination.
         optimizeGroupBy(select_query, source_columns_set, context);
 
+        /// Eliminate min/max/any aggregators of functions of GROUP BY keys
+        optimizeAggregateFunctionsOfGroupByKeys(select_query);
```

Comment:
We need a setting to disable the optimisation. It' possible to combine with some other setting that enables group by optimisation on make a new one.